ambry-commons/src/test/java/com.github.ambry.commons/MockHelixPropertyStore.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.I0Itec.zkclient.DataUpdater;
+import org.I0Itec.zkclient.IZkChildListener;
+import org.I0Itec.zkclient.IZkDataListener;
+import org.apache.helix.BaseDataAccessor;
+import org.apache.helix.store.HelixPropertyListener;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.zookeeper.data.Stat;
+
+
+/**
+ * A mock implementation of {@link HelixPropertyStore} and {@link BaseDataAccessor}.
+ */
+class MockHelixPropertyStore<T> implements HelixPropertyStore<T>, BaseDataAccessor<T> {
+  private final Map<String, T> pathToRecords = new HashMap<>();
+  private final Map<String, Set<HelixPropertyListener>> pathToListeners = new HashMap<>();
+
+  @Override
+  public void start() {
+    throw new IllegalStateException("Not implemented");
+  }
+
+  @Override
+  public void stop() {
+    throw new IllegalStateException("Not implemented");
+  }
+
+  @Override
+  public void subscribe(String parentPath, HelixPropertyListener listener) {
+    Set<HelixPropertyListener> listeners = pathToListeners.get(parentPath);
+    if (listeners == null) {
+      listeners = new HashSet<>();
+      pathToListeners.put(parentPath, listeners);
+    }
+    listeners.add(listener);
+  }
+
+  @Override
+  public void unsubscribe(String parentPath, HelixPropertyListener listener) {
+    Set<HelixPropertyListener> listeners = pathToListeners.get(parentPath);
+    if (listeners != null) {
+      listeners.remove(listener);
+    }
+  }
+
+  @Override
+  public boolean create(String path, T record, int options) {
+    throw new IllegalStateException("Not implemented");
+  }
+
+  @Override
+  public boolean set(String path, T record, int options) {
+    System.out.println("Setting to store path: " + path + ", record: " + record.toString());
+    return setAndNotify(path, record);
+  }
+
+  @Override
+  public boolean set(String path, T record, int expectVersion, int options) {
+    throw new IllegalStateException("Not implemented");
+  }
+
+  @Override
+  public boolean update(String path, DataUpdater<T> updater, int options) {
+    T newRecord = updater.update(pathToRecords.get(path));
+    System.out.println("Updating to store path: " + path + ", record: " + newRecord.toString());
+    return setAndNotify(path, newRecord);
+  }
+
+  @Override
+  public boolean remove(String path, int options) {
+    if (path.equals("/")) {
+      notifyListeners("/", HelixPropertyStoreUtils.StoreOperationType.DELETE);
+      pathToRecords.clear();
+      return true;
+    } else {
+      throw new IllegalStateException("Not implemented");
```

Comment:
Checking `/` means checking the existence of a store. This is the only needed function by far, so there is no such a case to remove a sub path in a store. Once `HelixStoreOperator` is lifted to the main side, code and test will be added accordingly.